crates/doc/src/parser/comment.rs

```diff
@@ -21,6 +21,7 @@ pub enum CommentTag {
     /// Copies all missing tags from the base function (must be followed by the contract name)
     Inheritdoc,
     /// Custom tag, semantics is application-defined
+    Variant,
```

Comment:
Every element of this enum is one of the natspec tags defined in the specs [here](https://docs.soliditylang.org/en/latest/natspec-format.html#tags). And `variant` isn't one of them.

Instead, is it not possible to parse this as `Custom("variant")` and do the filtering in the writer.